app/javascript/schools/courses/components/CourseEditor.res

```diff
@@ -48,6 +51,7 @@ type state = {
   filter: filter,
   totalEntriesCount: int,
   relaodCourses: bool,
+  singleCourse: option<Course.t>,
```

Comment:
We are using `selectedCourse` for managing another state. I will use  `currentCourse` here.

Reply:
@prashanth726 Why would there be a `selectedCourse` and `currentCourse` in the same component? That doesn't make sense. 😕

Also, note the typo in the line above the one highlighted here - `relaodCourses` - this should be fixed.

Reply:
@prashanth726 : Just tried digging deep into the presence of `selectedCourse` prop in the component and figured out that the actual bug was caused by two things:
- Wrong scope in the policy - was loading only the active courses in the school (which is fixed now)
- The `selectedCourse` which was an optional prop of the component was not getting decoded properly though the data was sent using the presenter. The reason being fields `highlights` and `public_preview` were missing in the presenter.
The client side code was correctly adding the `selectedCourse` into the list of courses and performing the search to display the editor. The absence of `selectedCourse` because of the decode issue was actually causing the bug.
https://github.com/pupilfirst/pupilfirst/blob/53a1228e4aa70812827cca6e75623da4d997acb2/app/javascript/schools/courses/components/CourseEditor.res#L466

Now that we have moved to the GraphQL approach , the presence of types will eliminate such issues in future. So let's cleanup the `selectedCourse` prop and its usage and have just a `selectedCourse` in the state.